tracing/opencensus.go

```diff
@@ -130,17 +138,22 @@ func WithExporter(name string, logger *zap.SugaredLogger) ConfigOption {
 			}
 			exporter = exp
 		case config.Zipkin:
-			// If name isn't specified, then zipkin.NewEndpoint will return an error saying that it
+			// If hostPort isn't specified, then zipkin.NewEndpoint will return an error saying that it
 			// can't find the host named ''. So, if not specified, default it to this machine's
 			// hostname.
-			if name == "" {
+			if hostPort == "" {
 				n, err := os.Hostname()
 				if err != nil {
 					return fmt.Errorf("unable to get hostname: %w", err)
 				}
-				name = n
+				hostPort = n
+			}
+			if name == "" {
+				name = hostPort
+			}
+			if strings.IndexByte(hostPort, ':') < 0 {
```

Comment:
Why not `!strings.Contains(hostPort, ":")`?

Reply:
If you're worried about performance: It does the exact same thing :). It optimizes `len(substr) == 1` to an IndexByte call.

Reply:
Also `LastIndex`?